win32/src/winapi/kernel32/resource.rs

```diff
@@ -112,25 +116,47 @@ pub fn FindResourceW(
     hModule: HMODULE,
     lpName: ResourceKey<&Str16>,
     lpType: ResourceKey<&Str16>,
-) -> u32 {
+) -> HRSRC {
     match find_resource(
         &machine.state.kernel32,
         machine.mem(),
         hModule.to_raw(),
         lpType,
         lpName,
     ) {
-        None => 0,
-        Some(mem) => machine.mem().offset_of(mem.as_ptr()),
+        None => HRSRC::null(),
+        Some(mem) => {
+            let handle = ResourceHandle(hModule.clone(), mem);
+            machine.state.kernel32.resource_handles.add(handle)
+        },
     }
 }
 
 #[win32_derive::dllexport]
-pub fn LoadResource(_machine: &mut Machine, hModule: u32, hResInfo: u32) -> u32 {
-    hResInfo
+pub fn LoadResource(machine: &mut Machine, hModule: HMODULE, hResInfo: HRSRC) -> u32 {
+    hResInfo.to_raw()
 }
 
 #[win32_derive::dllexport]
-pub fn LockResource(_machine: &mut Machine, hResData: u32) -> u32 {
-    hResData
+pub fn LockResource(machine: &mut Machine, hResData: HRSRC) -> u32 {
+    let handle = match machine.state.kernel32.resource_handles.get(hResData) {
+        None => return 0,
+        Some(handle) => handle,
+    };
+
+    let mem = machine.mem();
+    let hInstance = handle.0.to_raw();
+    let image = mem.slice(hInstance..);
+
+    let resource = &image[handle.1.clone()];
+
+    machine.mem().offset_of(resource.as_ptr())
```

Comment:
If I followed this math, this could be just `hInstance + handle.1`, right?

Reply:
Just to get all the pointers straight:
- pe::find_resource returns a range that is within the image
- maybe kernel32::find_resource should return a range that is within the process memory (so with hInstance already added)?
- so then this would only need to return handle.1?